calico/felix/felix.py

```diff
@@ -159,29 +160,36 @@ def _main_greenlet(config):
 
 
 def main():
-    try:
-        # Initialise the logging with default parameters.
-        common.default_logging()
+    # Initialise the logging with default parameters.
+    common.default_logging()
 
-        # Create configuration, reading defaults from file if it exists.
+    # Create configuration, reading defaults from file if it exists.
+    try:
+        parser = optparse.OptionParser()
+        parser.add_option('-c', '--config-file', dest='config_file',
+                          help="configuration file to use",
+                          default="/etc/calico/felix.cfg")
+        options, args = parser.parse_args()
+        config = Config(options.config_file)
+    except Exception:
+        # Config loading error, and not just invalid parameters (from optparse)
+        # as they generate a SystemExit. Attempt to open a log file, ignoring
+        # any errors it gets, before we raise the exception.
         try:
-            config = Config("/etc/calico/felix.cfg")
-        except:
-            # Attempt to open a log file, ignoring any errors it gets, before
-            # we raise the exception.
-            try:
-                common.complete_logging("/var/log/calico/felix.log",
-                                        logging.DEBUG,
-                                        logging.DEBUG,
-                                        logging.DEBUG)
-            except:
-                pass
+            common.complete_logging("/var/log/calico/felix.log",
+                                    logging.DEBUG,
+                                    logging.DEBUG,
+                                    logging.DEBUG)
+        except Exception:
+            pass
 
-            raise
+        raise
```

Comment:
What is this raise statement for? Raising the exception here will, in practice, kill the process, so having gone to all the effort of completing the logging setup is a waste of time, *especially* as the exception isn't logged out explicitly anyway.